src/Filter/StringFilter.php

```diff
@@ -114,8 +122,28 @@ public function getRenderSettings()
         ]];
     }
 
-    private function getOperator(int $type): string
+    private function getOperator(int $type, string $value): string
     {
+        // using an explicit comparison is preferred over LIKE
+        if ('' === $value) {
+            return StringOperatorType::TYPE_NOT_CONTAINS === $type ? '<>' : '=';
```

Comment:
I'm not ok with this.

CONTAINS '' should return everything.
EQUAL '' should return = ''.
NOT CONTAINS '' should return nothing.